utilities/storage.py

```
@@ -109,76 +110,136 @@ def create_dummy_first_consumer_pod(volume_mode=DataVolume.VolumeMode.FILE, dv=N
-        source_ref=source_ref,
-    ) as dv:
-        if sc_volume_binding_mode_is_wffc(sc=storage_class, client=client) and consume_wffc:
-            create_dummy_first_consumer_pod(dv=dv)
-        yield dv
-    utilities.artifactory.cleanup_artifactory_secret_and_config_map(
-        artifactory_secret=artifactory_secret, artifactory_config_map=cert_created
-    )
+    try:
+        if source_dict is None and source_ref is None:
+            if not source:
+                raise ValueError("'source' is required when 'source_dict' and 'source_ref' are not provided")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)